doc/api/webstreams.md

```diff
@@ -0,0 +1,1096 @@
+<!-- YAML
+added: REPLACEME
+-->
+
+* `destination` {WritableStream} A {WritableStream} to which this
+  `ReadableStream`'s data will be written.
+* `options` {Object}
+  * `preventAbort` {boolean} When `true`, errors in this `ReadableStream`
+    will not cause `transform.writable` to be aborted.
+  * `preventCancel` {boolean} When `true`, errors in the destination
+    `transform.writable` is not cause this `ReadableStream` to be
+    canceled.
+  * `preventClose` {boolean} When `true`, closing this `ReadableStream`
+    will no cause `transform.writable` to be closed.
+  * `signal` {AbortSignal} Allows the transfer of data to be canceled
+    using an {AbortController}.
+* Returns: A promise fulfilled with `undefined`
+
+Causes the `readableStream.locked` to be `true` while the pipe operation
+is active.
+
+#### `readableStream.tee()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {ReadableStream[]}
+
+Returns a pair of new {ReadableStream} instances to which this
+`ReadableStream`'s data will be forwarded. Each will receive the
+same data.
+
+Causes the `readableStream.locked` to be `true`.
+
+#### `readableStream.values([options])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `options` {Object}
+  * `preventCancel` {boolean} When `true`, prevents the {ReadableStream}
+    from being closed when the async iterator abruptly terminates.
+    **Defaults**: `false`
+
+Creates and returns an async iterator usable for consuming this
+`ReadableStream`'s data.
+
+Causes the `readableStream.locked` to be `true` while the async iterator
+is active.
+
+```mjs
+const stream = new ReadableStream(getSomeSource());
+
+for await (const chunk of stream.values({ preventCancel: true }))
+  console.log(Buffer.from(chunk).toString());
+```
+
+#### Async Iteration
+
+The {ReadableStream} object supports the async iterator protocol using
+`for await` syntax.
+
+```mjs
+const stream = new ReadableStream(getSomeSource());
+
+for await (const chunk of stream)
+  console.log(Buffer.from(chunk).toString());
+```
+
+The async iterator will consume the {ReadableStream} until it terminates.
+
+By default, if the async iterator exits early (via either a `break`,
+`return`, or a `throw`), the {ReadableStream} will be closed. To prevent
+automatic closing of the {ReadableStream}, use the `readableStream.values()`
+method to acquire the async iterator and set the `preventCancel` option to
+`true`.
+
+The {ReadableStream} must not be locked (that is, it must not have an existing
+active reader). During the async iteration, the {ReadableStream} will be locked.
+
+#### Transfering with `postMessage()`
+
+A {ReadableStream} instance can be transferred using a {MessagePort}.
+
+```js
+const stream = new ReadableStream(getReadableSourceSomehow());
+
+const { port1, port2 } = new MessageChannel();
+
+port1.onmessage = ({ data }) => {
+  data.getReader().read().then((chunk) => {
+    console.log(chunk);
+  });
+};
+
+port2.postMessage(stream);
```

Comment:
To transfer a stream (or really *any* transferable object), you need to add it to the `transferList`.
```suggestion
port2.postMessage(stream, [stream]);
```

Reply:
Good catch :-)

Reply:
Note that the modern easier-to-read syntax (not sure if Node supports it yet) is `postMessage(stream, { transfer: [stream] })`